thrift/thrift-gen/compile_test.go

```diff
@@ -115,6 +115,45 @@ func TestExternalTemplate(t *testing.T) {
 	}
 }
 
+func TestExternalTemplateWithGoType(t *testing.T) {
```

Comment:
Too much code duplication. Just change the above test to range over the methods and add that to the verification rather than duplicating the whole test.